pkg/cmd/cli/cmd/process.go

```diff
@@ -115,15 +118,31 @@ func RunProcess(f *clientcmd.Factory, out io.Writer, cmd *cobra.Command, args []
 		mapping *meta.RESTMapping
 	)
 
+	sourceNamespace := namespace
+
 	version, kind, err := mapper.VersionAndKindForResource("template")
 	if mapping, err = mapper.RESTMapping(kind, version); err != nil {
 		return err
 	}
 
-	// When storedTemplate is not empty, then we fetch the template from the
+	// When templateName is not empty, then we fetch the template from the
 	// server, otherwise we require to set the `-f` parameter.
-	if len(storedTemplate) > 0 {
-		templateObj, err := client.Templates(namespace).Get(storedTemplate)
+	if len(templateName) > 0 {
+		var (
+			storedTemplate, rs string
+			ok                 bool
+		)
+		sourceNamespace, rs, storedTemplate, ok = parseNamespaceResourceName(storedTemplate)
```

Comment:
don't you need to handle len(sourceNamespace) == 0 by using the default namespace?